tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/requestbuilders/GenerateUrlRequestBuilders.java

```diff
@@ -0,0 +1,68 @@
+package ca.bc.gov.open.jagefilingapi.qa.requestbuilders;
+
+import ca.bc.gov.open.jagefilingapi.qa.backend.generateurlpayload.GenerateUrlPayload;
+import ca.bc.gov.open.jagefilingapi.qa.backendutils.APIResources;
+import ca.bc.gov.open.jagefilingapi.qa.backendutils.TestUtil;
+import ca.bc.gov.open.jagefilingapi.qa.frontendutils.JsonDataReader;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import java.io.IOException;
+import static io.restassured.RestAssured.given;
+
+public class GenerateUrlRequestBuilders {
+
+    private  RequestSpecification request;
+    private static final String X_AUTH_USER_ID = "X-Auth-UserId";
+
+    private  GenerateUrlPayload payloadData;
+
+    public Response requestWithValidCSOAccountGuid(String resourceValue) throws IOException {
+        payloadData = new GenerateUrlPayload();
+        APIResources resourceAPI = APIResources.valueOf(resourceValue);
+
+        String validExistingCSOGuid = JsonDataReader.getCsoAccountGuid().getValidExistingCSOGuid();
+
+        request = given().spec(TestUtil.requestSpecification()).header(X_AUTH_USER_ID,validExistingCSOGuid).body(payloadData.generateUrlFinalPayload());
+        return request.when().post(resourceAPI.getResource()).then().spec(TestUtil.responseSpecification()).extract().response();
+    }
+
+    public Response requestWithNonExistingCSOAccountGuid(String resourceValue) throws IOException {
+        payloadData = new GenerateUrlPayload();
+        APIResources resourceAPI = APIResources.valueOf(resourceValue);
+
+        String  nonExistingCSOGuid = JsonDataReader.getCsoAccountGuid().getNonExistingCSOGuid();
```

Comment:
nit/question: looks like there is an extra space here after String
might just be github tho :)

Reply:
Good eyes!fixed. Thanks.